src/dataframe/dataframe.jl

```diff
@@ -107,11 +108,15 @@ struct DataFrame <: AbstractDataFrame
         minlen, maxlen = extrema(lengths)
         if minlen != maxlen || minlen == maxlen == 1
             # recycle scalars
-            for i in 1:length(columns)
-                isa(columns[i], AbstractArray) && continue
-                columns[i] = fill!(Tables.allocatecolumn(typeof(columns[i]), maxlen),
-                                   columns[i])
-                lengths[i] = maxlen
+            for (i, col) in enumerate(columns)
+                if (col isa AbstractArray && ndims(col) == 0) || col isa Ref
```

Comment:
Maybe use the same construct here?
```suggestion
                if col isa Union{AbstractArray{<:Any, 0}, Ref}
```